src/reconcile/rec_visibility.c

```diff
@@ -322,6 +322,16 @@ __wt_rec_upd_select(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_INSERT *ins, v
         return (0);
     }
 
+    if (F_ISSET(r, WT_REC_VISIBILITY_ERR))
+        WT_PANIC_RET(session, EINVAL, "reconciliation error, update not visible");
```

Comment:
Shouldn't this check for `has_newer_updates`?  I'm trying to follow the flow through here, and it looks like we can get here when all the updates are stable.  What prevents that?

Reply:
I was trying to return as early as possible. You are right, this is not correct.

Reply:
We should check `has_newer_updates`.